reference/forms/types/options/translation_domain.rst.inc

```diff
@@ -1,7 +1,7 @@
 translation_domain
 ~~~~~~~~~~~~~~~~~~
 
-**type**: ``string`` **default**: ``messages``
+**type**: ``string`` or ``false``  **default**: ``messages``
```

Comment:
Nice catch! Could you please also change the default which is actually `null`.

You should also note that `null` uses the parent form's value, `messages` is the default for the translator, that does not concerns forms.